src/ThreadedSensitivities.jl

```diff
@@ -1,3 +1,36 @@
 import SciMLBase: ODESolution, build_solution, LinearInterpolation
 using Sundials
 using Base.Threads
+"""
+generate individual sensitivity ODEs for each parameter
+"""
+function generatesensitivityodes(react,sol)
+    sa_list = []
+    y0 = react.y0
+    tspan = react.tspan
+    p = react.p
+    for i in 1:length(p)
+        r = deepcopy(react)
```

Comment:
Why do we need to deepcopy this? Is the reactor changed somehow?

Reply:
This deepcopy is the fix for the bug that took Mark and I so long to find. The issue is that some of our domains precompute kfs and krevs, but in gas phase these arrays need modified to account for efficiencies. So if you don't copy  it you have multiple threads trying to modify the same array. It's definitely possible to reduce the copy time by being more clever and careful, but there's some caveats particularly related to our use of SparseArrays that make it non-trivial to do that.